src/Api/TMDB/tmdbService.tsx

```diff
@@ -0,0 +1,91 @@
+export const fetchMovieGenres = async () => {
+    const options = {
+        method: "GET"
+    }
+    const url = `${import.meta.env.VITE_TMDB_URL}genre/movie/list?api_key=${import.meta.env.VITE_TMDB_APIKEY}`;
+    const response = await fetch(url, options);
+    const responseData = await response.json();
+    return { status: response.status, data: responseData };
+}
+
+export const fetchMovies = async (pageNumber: number) => {
+    const options = {
+        method: "GET"
+    }
+    const url = `${import.meta.env.VITE_TMDB_URL}movie/top_rated?api_key=${import.meta.env.VITE_TMDB_APIKEY}&page=${pageNumber}`;
+    const response = await fetch(url, options);
+    const responseData = await response.json();
+    return { status: response.status, data: responseData };
+}
```

Comment:
if you look at your previous function and this one, you'll realize 80% of it  is the same code. You can refactor it and create a new function that just takes the url and other params and does everything else internally. Example
- setting up the options
- firing the request
- extracting the json object. 
Do not repeat yourself

Reply:
Created a function "fetchGetRequest" which will perform 

> setting up the options
> firing the request
> extracting the json data